<launch>
  <arg name="stream_rate"         default="100"   />
  <arg name="logging_start_delay" default="5"     />
  <arg name="do_logging"          default="true" />
  <arg name="arduino_pins"        default="[]"    />
  <arg name="rcin_pins"           default="[]"    />
  <arg name="rcout_pins"          default="[]"    />

  <rosparam param="stream_rate"  subst_value="true">$(arg stream_rate) </rosparam>
  <rosparam param="arduino_pins" subst_value="true">$(arg arduino_pins)</rosparam>
  <rosparam param="rcin_pins"    subst_value="true">$(arg rcin_pins)   </rosparam>
  <rosparam param="rcout_pins"   subst_value="true">$(arg rcout_pins)  </rosparam>

  <node pkg="mavros" type="mavros_node" name="mavros" output="screen" respawn="false"/>

  <node pkg="rosservice" type="rosservice" name="set_all_stream_rate"      args="call --wait /mavros/set_stream_rate 0 0 1"                   launch-prefix="bash -c 'sleep 1; $0 $@' "/>
  <node pkg="rosservice" type="rosservice" name="set_odom_stream_rate"     args="call --wait /mavros/set_stream_rate 3 $(arg stream_rate) 1"  launch-prefix="bash -c 'sleep 2; $0 $@' "/>
  <node pkg="rosservice" type="rosservice" name="set_rc_stream_rate"       args="call --wait /mavros/set_stream_rate 6 $(arg stream_rate) 1"  launch-prefix="bash -c 'sleep 3; $0 $@' "/>
  <node pkg="rosservice" type="rosservice" name="set_airspeed_stream_rate" args="call --wait /mavros/set_stream_rate 11 $(arg stream_rate) 1" launch-prefix="bash -c 'sleep 4; $0 $@' "/>


  <node pkg="dfti2" type="pixhawk_publisher"    name="pixhawk_publisher" output="screen" respawn="false" launch-prefix="bash -c 'sleep 5; $0 $@' "/>
  <node pkg="dfti2" type="arduino_publisher.py" name="arduino_publisher" output="screen" respawn="false" launch-prefix="bash -c 'sleep 5; $0 $@' "/>

  <node if="$(arg do_logging)" name="dfti" pkg="dfti2" type="dfti" output="screen" launch-prefix="bash -c 'sleep $(arg logging_start_delay); $0 $@' "/>
</launch>
